### DISCUSSION TITLE
2.7.1

### DISCUSSION THREAD

**Discussion by markcheret** [markcheret/footnotes#104] [Releases]
The latest fix by @pewgeuges has been rolled into the pending release.

How does the planning for 2.7.1 look?
I put it together here https://github.com/markcheret/footnotes/milestone/3

- Achievable until next Friday or do we need more time?
- Any major changes in there that should be moved to 2.8?

**Comment by markcheret** [comment]
@Rumperuu planned this:

> #36 plan on doing in conjunction with #35 which in turn depends on #43 (as type casts and declarations require PHP 7+).
> 
> #33 isn't too pressing as we're not renaming any files at the moment; it's a potential gotcha that might catch us out in the future though.
> #43 is probably the higher priority, as there have been a couple issues lately revolving around older PHP versions so it will be good to actually have a formal idea of which we do and don't support.

**Comment by markcheret** [comment]
@lolzim  is on #103

**Comment by markcheret** [comment]
We're ready to test and release.
I have no knowledge on how to minify js and css. So a minification run is still required I guess, before tagging and pushing to WP, right?

https://github.com/markcheret/footnotes/milestone/3

**Reply by lolzim** [reply]
I have no idea either :-/

**Reply by Rumperuu** [reply]
~~I had to minify the stylesheets [manually](https://cssminifier.com/). The pre-release is now out on [the GitHub repo.](https://github.com/markcheret/footnotes/releases/tag/2.7.1) and [the SVN repo.](https://plugins.svn.wordpress.org/footnotes/trunk/), done via the release helper script — Stable Tag is still 2.7.0.~~

~~The already-existing [2.7.1 tag](https://plugins.svn.wordpress.org/footnotes/tags/2.7.1/) will need to be deleted so we can push this proper 2.7.1 to `tags/`.~~

Cancel that, the SVN version hasn't uploaded correctly.

**Reply by lolzim** [reply]
Yes, it misses the class/tasks.php file at least.

**Reply by Rumperuu** [reply]
[Fixed](https://plugins.svn.wordpress.org/footnotes/trunk/) - download the archive from the WP Plugin Directory [here](https://downloads.wordpress.org/plugin/footnotes.zip).

Part of the issue was that `svn ci` doesn't commit file deletions, so all of the deleted files were staying and throwing fatal errors because they were looking for the now-renamed `MCI_*` classes. I've now fixed the SVN deletion in the release helper in #127.

I'm not sure why the first time around all of the files under `class/` weren't changed. Hopefully the process will have been nailed down enough by 2.8 that whatever it was will be sorted.

**Reply by lolzim** [reply]
OK, got it in my test WP. It seems OK.

NB: I saw this after uploading the .zip:
<img width="673" alt="image" src="https://user-images.githubusercontent.com/20860210/115291659-4eb97080-a155-11eb-826e-a6252b30aed9.png">

Shouldn't it show the min WP & PHP versions requested?

And, once installed, it knows 'em:
<img width="233" alt="image" src="https://user-images.githubusercontent.com/20860210/115291881-904a1b80-a155-11eb-8f0f-4330d4b74b04.png">

🤨

**Comment by markcheret** [comment]
I just saw the commit emails on SVN..
Thanks for the work and bugfixing that went along with the SVN monstrosity to get it to cooperate.

Are we good now to tag it all for 2.7.1 on WP?

I'm going to download 2.7.1 from GH now and check it out.

What's the process going to be?

**Reply by Rumperuu** [reply]
The GitHub release [failed](https://github.com/markcheret/footnotes/actions/runs/764655307) - the fix (I'm hoping) is in #127 

Best to download the ‘Development Version’ (== `trunk/`) archive from the SVN [here](https://downloads.wordpress.org/plugin/footnotes.zip).

Once we're happy that everything is working fine, it should just be a case of changing the ‘Stable Tag’ field in `trunk/readme.txt` to 2.7.1. Then the Plugin Directory will automatically parse it, copy the contents of `trunk/` to `tags/2.7.1/` and tell the users that there is a new update available.

There's no rush though; the milestone due date isn't until Friday, so we can leave it sat in `trunk/` until then in case we come up with any more bugfixes between now and then that we can then roll into 2.7.1.

**Reply by Rumperuu** [reply]
GitHub release [passed](https://github.com/markcheret/footnotes/actions/runs/765216325) - archive available from https://github.com/markcheret/footnotes/releases/tag/2.7.1

I'm not *really* sure what the use case of having a release archive here as well as on the SVN repo. actually is, but it's definitely **neat**

**Reply by markcheret** [reply]
An off-platform backup is _always_ neat. Wish I'd known that like 10 years ago. I lost 3 whole self-produced podcasts (40 episodes each) to ignorance of the aforementioned neatness.

**Comment by lolzim** [comment]
On my main WP site, it's still 2.7.0 and no new release available; normal?

**Reply by Rumperuu** [reply]
Yep, Stable Tag is still 2.7.0 - see https://github.com/markcheret/footnotes/discussions/104#discussioncomment-632545

**Reply by lolzim** [reply]
OK, I thought it was already fixed and released, sorry :)
May I help?

**Reply by Rumperuu** [reply]
Just poke around and make sure nothing's broken. IIRC you posted somewhere that the WP version was showing up incorrectly (but I can't find the comment now)?

**Reply by lolzim** [reply]
Can't found it here, but I have the screenshots in my pasteboard manager :)

When I updated to .zip, WP showed:
<img width="673" alt="image" src="https://user-images.githubusercontent.com/20860210/115630737-d42f5300-a304-11eb-9fdb-af9d3644b1bf.png">

After update, it shows it knows those versions:
<img width="233" alt="image" src="https://user-images.githubusercontent.com/20860210/115630767-e4473280-a304-11eb-9c32-a9493f73482f.png">

Question is: why couldn't it show required PHP & WP versions after uploading .zip?

**Reply by Rumperuu** [reply]
How are you viewing that screen? If I install the Plugin as a `.zip` I just see this, no mention of WP/PHP versions:

![Screenshot from 2021-04-22 07-44-17](https://user-images.githubusercontent.com/5621904/115670641-8c7aec80-a341-11eb-8d04-aea5a4e2d274.png)

**Reply by lolzim** [reply]
Hi,

Here's the path:

Plugin add
<img width="221" alt="image" src="https://user-images.githubusercontent.com/20860210/115691619-83e4df00-a35e-11eb-9670-ec2e541063dd.png">

Upload
<img width="436" alt="image" src="https://user-images.githubusercontent.com/20860210/115691691-92cb9180-a35e-11eb-8ca0-2856353974dd.png">

Choose zip
<img width="138" alt="image" src="https://user-images.githubusercontent.com/20860210/115691896-bdb5e580-a35e-11eb-87c1-8cb5584a7047.png">

Install
<img width="176" alt="image" src="https://user-images.githubusercontent.com/20860210/115691923-c7d7e400-a35e-11eb-9b32-c6ef0dde4d1e.png">

and then:
<img width="663" alt="image" src="https://user-images.githubusercontent.com/20860210/115691945-ce665b80-a35e-11eb-9a3e-292c994fd3fd.png">

cu

**Reply by Rumperuu** [reply]
How strange, here is me doing the exact same steps (and using the `.zip` from the ‘Development Version’ selector on the WP Plugin Directory);

![Screenshot from 2021-04-22 22-00-47](https://user-images.githubusercontent.com/5621904/115784745-6e030880-a3b6-11eb-9c7d-380d162e35cb.png)

![Screenshot from 2021-04-22 22-01-16](https://user-images.githubusercontent.com/5621904/115784746-6e9b9f00-a3b6-11eb-9557-589b49f563e4.png)

![Screenshot from 2021-04-22 22-01-21](https://user-images.githubusercontent.com/5621904/115784749-6f343580-a3b6-11eb-9932-6d0c18b5869b.png)

![Screenshot from 2021-04-22 22-01-30](https://user-images.githubusercontent.com/5621904/115784751-6fcccc00-a3b6-11eb-8968-18d49e8d7410.png)

Do you have another Plugin that shows pre-install info on new Plugins running, or anything like that?

**Reply by lolzim** [reply]
This blog does have the plugin before uploading the zip file? It seems not, and it is the case on my.
Here's the full view:
![image](https://user-images.githubusercontent.com/20860210/115788037-5e3df100-a3c3-11eb-8a20-778b4d234656.png)
(NB: as the plugin is already installed, it asks if this zip should replace the existing one)

And no, no other plugin in this WP dedicated to fn tests:
![image](https://user-images.githubusercontent.com/20860210/115788055-672ec280-a3c3-11eb-9113-34c5090582fb.png)

**Reply by Rumperuu** [reply]
That was it - I didn't have it installed so I wasn't seeing that screen (which seems pretty obvious in retrospect :disappointed:)

Should be fixed in #139

**Comment by Rumperuu** [comment]
2.7.1 is now out on the SVN repo: https://plugins.svn.wordpress.org/footnotes/trunk/

In theory, at some point it will parse the new Stable Tag in `trunk/readme.txt` and automatically create a new tag. Fingers crossed.

**Reply by markcheret** [reply]
that would be new to me. In the past I always had to push a copy of the stable trunk to the desired tags folder.

**Reply by Rumperuu** [reply]
Ah yeah you're right, I misinterpreted what had happened in #55 (I think Pew made the tag, but I thought it was because the Plugin Directory had parsed the `trunk/readme.txt` and done it automatically). I've copied the 2.7.1 tag now.

**Comment by markcheret** [comment]
Congratulations team, this feels like the biggest release yet!
I'm very happy the process works and we had an orderly release.
I'm in party mode right now and will BBQ a steak with fries and open a cold brew in your honour.

**Reply by Rumperuu** [reply]
[Spoke too soon](https://wordpress.org/support/topic/how-to-re-hide-accessibility-captions-for-sighted-users/) 😭 

The entire stylesheet build process is a complete mess. There are also some new AMP-related ones that have snuck into the `css/` dir without being added to the build process (presumably at some point during the game of musical repos).

I've reverted the Stable Tag to 2.7.0, which should stop people who haven't already done so from being prompted to upgrade the Plugin. I'm not sure if people who *have* upgraded will be prompted to downgrade or not.

Pausing all releases until I've have a chance to do #134 (at least), which should be this weekend.

**Reply by markcheret** [reply]
there will be no downgrade option provided

**Reply by Rumperuu** [reply]
Following @pewgeuges's [comment](https://wordpress.org/support/topic/how-to-re-hide-accessibility-captions-for-sighted-users/#post-14357950), I've checked and yes the `.footnotes_table caption.accessibility` CSS rule has been included in the minified stylesheets since 2.6.0, up *until* 2.7.1.

The issue appears to be that copies of (some of) the minified stylesheets in `css/tmp/` are being [tracked](https://github.com/markcheret/footnotes/tree/main/css/tmp), and that these files are a month old (i.e., presumably just predating 2.6.0) and do not include the `.footnotes_table caption.accessibility` (nor, presumably, any of the other stylesheet-related fixes introduced since 2.6.0). [`gitignore` is correctly set](https://github.com/markcheret/footnotes/blob/main/.gitignore) to ignore any `tmp/` directories, but I guess that doesn't apply to files that are already set to be tracked. At some point in the release process, they must have overridden the newly-generated concatenated + minified stylesheets.

Downgrading to 2.7.0 will solve any issues (I've checked and the 2.7.0 stylesheets are definitely up-to-date on the SVN repo.), and I'm going to stand by my comment about not releasing anything else until I've be able to take full stock of just what on *Earth* is going on in that `css/` directory.

**Reply by markcheret** [reply]
OK. Thanks for that. Also, I guess the SVN stuff is an animal of its own. We might need to purge everything before committing and making SVN believe all is new.

**Comment by Rumperuu** [comment]
Post-mortem discussion in #158
